django/core/management/__init__.py

```diff
@@ -177,8 +177,11 @@ def fetch_command(self, subcommand):
         try:
             app_name = commands[subcommand]
         except KeyError:
-            # This might trigger ImproperlyConfigured (masked in get_commands)
-            settings.INSTALLED_APPS
+            if os.environ.get('DJANGO_SETTINGS_MODULE'):
+                # This might trigger ImproperlyConfigured (masked in get_commands)
```

Comment:
If it's possible to improve this comment I think that would be a plus as I still have some trouble understanding it (and it seems no tests fail if you comment out `settings.INSTALLED_APPS`).